docs/source/preparing_datasets/dataset_format.md

```diff
@@ -0,0 +1,25 @@
+# Dataset Format
```

Comment:
utterly unmentioned is how the meta files work? what their entries look like in the index.json? how they can coexist, etc

suspect these docs are going to cause support traffic from people trying to 'bring their own' csv/jsonl/etc

Reply:
Mind including information about the `meta` files in a separate PR? the old docs don't have any information on that, and since you're assigned to this page and know the most about the format, you'd be best equipped to add a bit on it. I included a "Metadata" section that covers the `index.json` file, but not the `meta` section yet.

Reply:
Or, if you could provide a blurb about the `meta` files that I can include here, that would be great too.